Tests/OTelTests/Logging/OTelBatchLogProcessorTests.swift

```diff
@@ -0,0 +1,168 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@testable import Logging
+@_spi(Logging) import OTel
+import OTelTesting
+import XCTest
+
+final class OTelBatchLogProcessorTests: XCTestCase {
+    private let resource = OTelResource(attributes: ["service.name": "log_batch_processor_tests"])
+
+    func testBatchLogProcessorAccumulatesUntilQueueSize() async throws {
+        let exporter = OTelInMemoryLogRecordExporter()
+        let batchProcessor = OTelBatchLogRecordProcessor(
+            exporter: exporter,
+            configuration: OTelBatchLogRecordProcessorConfiguration(
+                environment: .detected(),
+                maximumQueueSize: 5,
+                scheduleDelay: .seconds(60) // Should never trigger
+            )
+        )
+
+        try await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            taskGroup.addTask(operation: batchProcessor.run)
+
+            let logHandler = OTelLogHandler(processor: batchProcessor, logLevel: .debug, resource: resource)
+            let logger = Logger(label: "Test", logHandler)
+
+            for i in 1...4 {
+                logger.info("\(i)")
+            }
+
+            // Records are emitted asynchronously, so checking this without delay
+            // is not representative
+            try await Task.sleep(for: .milliseconds(1))
+            XCTAssertEqual(exporter.records, [])
+
+            logger.info("5")
+
+            // Records are emitted asynchronously, so let's wait for that to happen
```

Comment:
Can we find a way to get rid of these sleep calls? I'm afraid it'll lead to flakiness. Perhaps the exporter could expose an async sequence we could call `next()` on inside the test?